src/zc/buildout/easy_install.py

```diff
@@ -399,11 +400,11 @@ def _call_easy_install(self, spec, ws, dest, dist):
             for d in dists:
                 newloc = os.path.join(dest, os.path.basename(d.location))
                 if os.path.exists(newloc):
-                    if os.path.isdir(newloc):
-                        shutil.rmtree(newloc)
-                    else:
-                        os.remove(newloc)
-                os.rename(d.location, newloc)
+                    logger.warn(
+                        "Path %s already exists.\nIf this contains a wrong "
+                        "package, please remove it yourself.", newloc)
+                else:
+                    os.rename(d.location, newloc)
```

Comment:
On Thu, Aug 18, 2016 at 2:19 PM, Leonardo Rochael Almeida
<notifications@github.com> wrote:
> This is a rare situation (when was the last time you saw an unpacked egg outside of the
> buildout be used as the find-links of a package in the buildout?), but if it happens, one
> buildout could see that an egg exists and attempt to use it before another buildout is
> finished copying it.

Rare, perhaps, but useful.  At Zope (back in the day), we used a
directory of built eggs as a cache when building packages for use by
OS packaging tools.  Reading something while someone else was writing
it wasn't a situation we ran into, but certainly could have happened
had we applied a little more parallelism (whether automated or human).


  -Fred

-- 
Fred L. Drake, Jr.    <fred at fdrake.net>
"A storm broke loose in my mind."  --Albert Einstein